science/gnuradio/Portfile

```diff
@@ -1,458 +1,295 @@
-    # CMAKE_C/CXX_STANDARD
+}
 
-    patchfiles-append \
-        patch-C_CXX_STANDARD.next.diff
+subport gnuradio-next {
 
-    # temporary patch to fix using Thrift 0.9.2 through 0.12.0
+    name            gnuradio-next
+    github.setup    gnuradio gnuradio 03d92ac19a11356175533aae4ad5804a354be075
+    version         20200419-[string range ${github.version} 0 7]
+    checksums       rmd160  0a9e1294036e786116c96d68f9116e3f2798d325 \
+                    sha256  bb2dab14e159a4b58ccf317ece3b4132959739f4b5b217d8e0add3c5ce581f5f \
+                    size    3950832
+    revision        0
 
+    long_description    ${description}: \
+        This port is kept up with the GNU Radio GIT 'master' \
+        branch (as of 2018-09-20), which is typically updated \
+        daily to weekly.  This version of GNU Radio represents \
+        the next major release of GNU Radio, and hence its API \
+        is likely to be different than that provided by gnuradio. \
+        This port may or not compile \
+        or function correctly, as it represents a work in progress. \
+        If it does not work, check back in a few days.  Or try \
+        deactivating the currently active gnuradio port, \
+        cleaning any current builds, and trying again.
+
+    conflicts           gnuradio gnuradio37
+
+    set python_versions { 3.6 3.7 3.8 }
+    set default_python_variant +python37
+
+    # fix Quartz behaviour
+    # TODO verify https://github.com/gnuradio/gnuradio/issues/2726
     patchfiles-append \
-        patch-fixes_for-thrift.next.diff
-
-    # temporary patch to fix usage of GNURadioControlPortClient
+        patch-quartz_drag_n_drop.next.diff \
+        patch-quartz_drop_crash.next.diff
 
-    patchfiles-append \
-        patch-fix-GNURadioControlPortClient-usage.next.diff
+    # set the version override string
+    set GR_VERSION_INFO "next-git-[string range ${github.version} 0 7](${version})"
 
 }
 
-# override githib PortGroup homepage setting
-
-homepage            https://www.gnuradio.org/
-
-use_parallel_build  yes
-
-# set Python version and variants
-set PythonVersionNoDot ""
-if {${subport} eq "gnuradio-next"} {
-
-    # select Python to use; default to 2.7
-
-    variant python27 conflicts python35 python36 python37 description {Build GNU Radio using Python 2.7} {}
-    variant python35 conflicts python27 python36 python37 description {Build GNU Radio using Python 3.5} {}
-    variant python36 conflicts python27 python35 python37 description {Build GNU Radio using Python 3.6} {}
-    variant python37 conflicts python27 python35 python36 description {Build GNU Radio using Python 3.7} {}
-
-    if {![variant_isset python27] &&
-        ![variant_isset python35] &&
-        ![variant_isset python36] &&
-        ![variant_isset python37]} {
-        default_variants +python27
+# obsolete 20191016
+subport gnuradio-devel {
+    replaced_by     gnuradio
+    version         3.8.0.0
+    PortGroup       obsolete 1.0
```

Comment:
To do a `replaced_by`, keep the version the same but do a rev-bump .

Reply:
my fault. thank you